machine-runner/src/runner/runner-internals.ts

```diff
@@ -211,26 +218,35 @@ export namespace RunnerInternals {
         const triggeringEvents = internals.queue
         internals.queue = []
 
-        const nextPayload = matchingReaction.handler(
-          {
-            self: internals.current.data.payload,
-          },
-          ...triggeringEvents,
-        )
-
-        internals.current = {
-          data: {
-            type: nextFactory.mechanism.name,
-            payload: nextPayload,
-          },
-          factory: nextFactory,
-        }
+        try {
+          const nextPayload = matchingReaction.handler(
+            {
+              self: internals.current.data.payload,
+            },
+            ...triggeringEvents,
+          )
 
-        internals.commandLock = null
+          internals.current = {
+            data: {
+              type: nextFactory.mechanism.name,
+              payload: nextPayload,
+            },
+            factory: nextFactory,
+          }
 
-        return {
-          executionHappened: true,
-          triggeringEvents,
+          internals.commandLock = null
+
+          return { type: PushEventTypes.React, triggeringEvents }
+        } catch (error) {
+          const failure = {
+            error,
+            current: internals.current.factory,
+            next: internals.current.factory,
+          }
+          return {
+            type: PushEventTypes.Failure,
+            failure: { ...failure },
```

Comment:
why make a copy of the `failure` object?

Reply:
it is an artifact from previous defensive measures.
not used anymore.
nice catch